src/auth/auth-config.ts

```diff
@@ -773,6 +792,61 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
         '"OIDCAuthProviderConfig.displayName" must be a valid string.',
       );
     }
+    if (typeof options.clientSecret !== 'undefined' &&
```

Comment:
why can't you just do if (!options.clientSecret)?

Reply:
I think (!options.clientSecret) is also correct.
I was just trying to follow the code style, I guess the reason here is that (!options.clientSecret) return true for clientSecret as empty string, null, false, {}, etc, while has overlap functionality with !validator.isNonEmptyString(options.clientSecret), but we only accept string type here, so here first check on whether clientSecret is defined could suffice.